sys/dev/cxgbe/tom/t4_cpl_io.c

```diff
@@ -1967,16 +1967,16 @@ alloc_aiotx_mbuf(struct kaiocb *job, int len)
 	 */
 	vm = job->userproc->p_vmspace;
 	map = &vm->vm_map;
-	start = (uintptr_t)job->uaiocb.aio_buf + job->aio_sent;
-	pgoff = start & PAGE_MASK;
+	start = __DEVOLATILE(char * __capability, job->uaiocb.aio_buf) + job->aio_sent;
```

Comment:
The fact that aio_buf is volatile is extremely suspect.......

Reply:
Oh that's a POSIX thing and because it's from way before any notion of atomics and concurrency was formalised in C... ew :(